evals/completion_fns/openai_inner_monologue.py

```diff
@@ -0,0 +1,67 @@
+"""
+Extending OpenAI Models with Chain-of-Thought Inner Monologue
+"""
+from evals import OpenAIChatCompletionFn, OpenAICompletionFn
+from evals.prompt.base import ChatCompletionPrompt, CompletionPrompt
+from evals.record import record_sampling
+
+
+class OpenAIInnerMonologueCompletionResult:
+    def __init__(self, response) -> None:
+        self.response = response
+
+    def get_completions(self) -> list[str]:
+        return [self.response.strip()]
+
+
+class OpenAIInnerMonologueCompletionFn:
```

Comment:
Should we name this something like `OpenAIChainOfThoughtCompletionFn`? Usually see this technique referred to as simply "chain of thought".

Reply:
+1 but also lol